src/editors/autocompleteEditor.js

```diff
@@ -77,247 +41,329 @@ function onBeforeKeyDown(event) {
-
-        if (indexOfMatch !== -1) {
-          match = cellValue.substr(indexOfMatch, _this.query.length);
-          cellValue = cellValue.replace(match, `<strong>${match}</strong>`);
-        }
-      }
-      TD.innerHTML = cellValue;
-    },
-    autoColumnSize: true,
-    modifyColWidth(width, col) {
-      // workaround for <strong> text overlapping the dropdown, not really accurate
-      const autoWidths = this.getPlugin('autoColumnSize').widths;
-      let columnWidth = width;
-
-      if (autoWidths[col]) {
-        columnWidth = autoWidths[col];
-      }
+  getValue() {
+    const selectedValue = this.rawChoices.find((value) => {
+      const strippedValue = this.stripValueIfNeeded(value);
+
+      return strippedValue === this.TEXTAREA.value;
+    });
 
-      return trimDropdown ? columnWidth : columnWidth + 15;
+    if (isDefined(selectedValue)) {
+      return selectedValue;
     }
-  });
 
-  // Add additional space for autocomplete holder
-  this.htEditor.view.wt.wtTable.holder.parentNode.style['padding-right'] = `${getScrollbarWidth() + 2}px`;
+    return this.TEXTAREA.value;
+  }
+
+  createElements(...args) {
+    super.createElements(...args);
 
-  if (skipOne) {
-    skipOne = false;
+    addClass(this.htContainer, 'autocompleteEditor');
+    addClass(this.htContainer, window.navigator.platform.indexOf('Mac') === -1 ? '' : 'htMacScroll');
   }
 
-  _this.instance._registerTimeout(() => {
-    _this.queryChoices(_this.TEXTAREA.value);
-  });
-};
+  prepare(...args) {
+    super.prepare(...args);
+  }
 
-AutocompleteEditor.prototype.queryChoices = function(query) {
-  this.query = query;
-  const source = this.cellProperties.source;
+  open(...args) {
+    this.instance.addHook('beforeKeyDown', onBeforeKeyDown);
+    // Ugly fix for handsontable which grab window object for autocomplete scroll listener instead table element.
+    this.TEXTAREA_PARENT.style.overflow = 'auto';
+    super.open(...args);
+    this.TEXTAREA_PARENT.style.overflow = '';
+
+    const choicesListHot = this.htEditor.getInstance();
+    const _this = this;
+    const trimDropdown = this.cellProperties.trimDropdown === void 0 ? true : this.cellProperties.trimDropdown;
+
+    this.showEditableElement();
+    this.focus();
+
+    choicesListHot.updateSettings({
+      colWidths: trimDropdown ? [outerWidth(this.TEXTAREA) - 2] : void 0,
+      width: trimDropdown ? outerWidth(this.TEXTAREA) + getScrollbarWidth() + 2 : void 0,
+      afterRenderer(TD, row, col, prop, value) {
+        const { filteringCaseSensitive, allowHtml } = _this.cellProperties;
+        let cellValue = stringify(value);
+        let indexOfMatch;
+        let match;
+
+        if (cellValue && !allowHtml) {
+          indexOfMatch = filteringCaseSensitive === true ? cellValue.indexOf(this.query) : cellValue.toLowerCase().indexOf(_this.query.toLowerCase());
```

Comment:
I noticed in this ternary the use of `this.query` in the first condition and `_this.query` in the second condition. Seems like it should consistently use `_this`?

Reply:
It should, looks like a mistake to me.